recipe/build-arrow.sh

```diff
@@ -62,6 +62,7 @@ cmake \
     -DARROW_BUILD_UTILITIES=OFF \
     -DBUILD_SHARED_LIBS=ON \
     -DARROW_DATASET=ON \
+    -DARROW_SUBSTRAIT=ON \
```

Comment:
Can we enable this for Windows as well? https://github.com/conda-forge/arrow-cpp-feedstock/blob/main/recipe/bld-arrow.bat#L17-L49